# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/RangeUtils.h
# Commit: 15ccf86bce5a
# Full Hash: 15ccf86bce5a786e4cbda785e186a2efd342b293
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-05-08 04:32:48
# Regressor Bug: 1932150
# File Overlap Count: 2
# Description:
#   Bug 1932150 - Fix some inconsistency about indicating whether the selection is allowed to cross the shadow boundary r=jjaschke,dom-core
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D246949
# ==============================================================================

diff -r 40f9fe959b19 -r 15ccf86bce5a dom/base/RangeUtils.h
--- a/dom/base/RangeUtils.h	Wed May 07 15:04:40 2025 +0000
+++ b/dom/base/RangeUtils.h	Wed May 07 15:05:08 2025 +0000
@@ -25,23 +25,29 @@
 struct ShadowDOMSelectionHelpers {
   ShadowDOMSelectionHelpers() = delete;
 
-  static nsINode* GetStartContainer(const AbstractRange* aRange,
-                                    bool aAllowCrossShadowBoundary);
+  static nsINode* GetStartContainer(
+      const AbstractRange* aRange,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 
-  static uint32_t StartOffset(const AbstractRange* aRange,
-                              bool aAllowCrossShadowBoundary);
+  static uint32_t StartOffset(
+      const AbstractRange* aRange,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 
-  static nsINode* GetEndContainer(const AbstractRange* aRange,
-                                  bool aAllowCrossShadowBoundary);
+  static nsINode* GetEndContainer(
+      const AbstractRange* aRange,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 
-  static uint32_t EndOffset(const AbstractRange* aRange,
-                            bool aAllowCrossShadowBoundary);
+  static uint32_t EndOffset(
+      const AbstractRange* aRange,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 
-  static nsINode* GetParentNodeInSameSelection(nsINode& aNode,
-                                               bool aAllowCrossShadowBoundary);
+  static nsINode* GetParentNodeInSameSelection(
+      const nsINode& aNode,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 
-  static ShadowRoot* GetShadowRoot(const nsINode* aNode,
-                                   bool aAllowCrossShadowBoundary);
+  static ShadowRoot* GetShadowRoot(
+      const nsINode* aNode,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 };
 }  // namespace dom
 